app/src/main/java/com/github/sdp/tarjetakuna/ui/browser/BrowserViewModel.kt

```diff
@@ -1,11 +1,54 @@
 package com.github.sdp.tarjetakuna.ui.browser
 
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import com.github.sdp.tarjetakuna.model.*
 
 class BrowserViewModel : ViewModel() {
 
-    val initialCards: ArrayList<MagicCard> = generateCards()
+    private val _searchState = MutableLiveData<String>()
+    private val _filterState = MutableLiveData<FilterState>()
+    private val _sorterState = MutableLiveData<Comparator<MagicCard>>()
+    private val _initialCards = MutableLiveData<ArrayList<MagicCard>>()
+    val searchState: MutableLiveData<String> = _searchState
```

Comment:
For these public val, you should create only _LiveData<A>_ and not _MutableLiveData_.
=> The fragment can only access immutable values from the viewmodel.